core/lib/multivm/src/versions/vm_fast/vm.rs

```diff
@@ -744,20 +782,23 @@ impl<S: ReadStorage, T: Tracer> World<S, T> {
 }
 
 impl<S: ReadStorage, T: Tracer> zksync_vm2::StorageInterface for World<S, T> {
-    fn read_storage(&mut self, contract: H160, key: U256) -> Option<U256> {
+    fn read_storage(&mut self, contract: H160, key: U256) -> StorageSlot {
         let key = &StorageKey::new(AccountTreeId::new(contract), u256_to_h256(key));
-        if self.storage.is_write_initial(key) {
-            None
-        } else {
-            Some(self.storage.read_value(key).as_bytes().into())
+        let value = U256::from_big_endian(self.storage.read_value(key).as_bytes());
+        let is_write_initial = self.storage.is_write_initial(key);
+        StorageSlot {
+            value,
+            is_write_initial,
```

Comment:
Maybe `has_been_written_in_this_block` or whatever descriptive name or comment would help the confusion around this.

Reply:
"is write initial" / "is initial" is used extensively throughout the codebase, so I don't think changing naming is really an option without changing those occurrences as well. I'll add a comment.